File: app/src/main/java/com/physphile/forbot/news/NewsFirebaseItem.java
package com.physphile.forbot.news;

public class NewsFirebaseItem {
    private String title;
    private String uri;
    private String text;
    private String author;
    private String date;
    private Integer number;
    private Integer mask;

    public NewsFirebaseItem() {
    }

    public NewsFirebaseItem(String title, String uri, String text, String author, String date, Integer number, Integer mask) {
        this.title = title;
        this.uri = uri;
        this.text = text;
        this.author = author;
        this.date = date;
        this.number = number;
        this.mask = mask;
    }

    public Integer getMask() {
        return mask;
    }

    public void setMask(Integer mask) {
        this.mask = mask;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getUri() {
        return uri;
    }

    public void setUri(String uri) {
        this.uri = uri;
    }

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public Integer getNumber() {
        return number;
    }

    public void setNumber(int number) {
        this.number = number;
    }
}

File: app/src/main/java/com/physphile/forbot/news/NewsFirebaseItem.kt
package com.physphile.forbot.news

class NewsFirebaseItem {
    var title: String? = null
    var uri: String? = null
    var text: String? = null
    var author: String? = null
    var date: String? = null
    var number: Int? = null
    var mask: Int? = null
    var coolDate: Long = 0

    constructor() {}

    private fun getLastNum(t: String?): Int {
        var s = t
        var ans = ""
        while (!s!!.isEmpty() && s[s.length - 1] != '.') {
            ans += s[s.length - 1]
            s = s.dropLast(1)
        }
        ans = ans.reversed()
        return ans.toInt()
    }

    private fun makeCoolDate() {
        var t = date
        val a = getLastNum(t)
        while (t!![t.length - 1] != '.') t = t.dropLast(1)
        t = t.dropLast(1)
        val b = getLastNum(t)
        while (t!![t.length - 1] != '.') t = t.dropLast(1)
        t = t.dropLast(1)
        val c = getLastNum(t)
        coolDate = -(c + b * 32 + a * 366).toLong()
    }

    constructor(title: String?, uri: String?, text: String?, author: String?, date: String?, number: Int?, mask: Int?) {
        this.title = title
        this.uri = uri
        this.text = text
        this.author = author
        this.date = date
        this.number = number
        this.mask = mask
        makeCoolDate()
    }

}